pkg/sac/for_resource_helpers.go

```diff
@@ -50,3 +51,55 @@ func (h ForResourceHelper) ReadAllowed(ctx context.Context, keys ...ScopeKey) (b
 func (h ForResourceHelper) WriteAllowed(ctx context.Context, keys ...ScopeKey) (bool, error) {
 	return h.AccessAllowed(ctx, storage.Access_READ_WRITE_ACCESS, keys...)
 }
+
+func (h ForResourceHelper) HasGlobalRead(ctx context.Context) (bool, error) {
+	return h.ReadAllowed(ctx)
+}
+
+func (h ForResourceHelper) HasGlobalWrite(ctx context.Context) (bool, error) {
+	return h.WriteAllowed(ctx)
+}
+
+func (h ForResourceHelper) FilterAccessibleNamespaces(
+	ctx context.Context,
+	accessMode storage.Access,
+	namespaces []*storage.NamespaceMetadata,
+) ([]*storage.NamespaceMetadata, error) {
+	filtered := make([]*storage.NamespaceMetadata, 0, len(namespaces))
```

Comment:
How about moving it just before the loop where it's used and use `slices.Clone(namespaces)` to explicitly show what we are doing

Reply:
That sounds good. I'll have a look.